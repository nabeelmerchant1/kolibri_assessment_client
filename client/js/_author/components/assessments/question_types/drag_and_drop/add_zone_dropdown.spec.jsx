import React            from 'react';
import { shallow }      from 'enzyme';
import AddZoneDropdown  from './add_zone_dropdown';

describe('drag_and_drop component', () => {
  let props;
  let result;
  let toggled;
  let addedRegion;
  let addedImage;

  beforeEach(() => {
    addedRegion = false;
    addedImage = false;
    toggled = false;
    props = {
      active: false,
      text: 'Spec',
      toggle: () => { toggled = true; },
      addByRegion: () => { addedRegion = true; },
      addByImage: () => { addedImage = true; },
    };
    result = shallow(<AddZoneDropdown {...props} />);
  });

  it('renders the component as a snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('toggles on click', () => {
    expect(toggled).toBeFalsy();
    const button = result.find('.au-c-btn--dropdown');
    button.simulate('click');
    expect(toggled).toBeTruthy();
  });

  it('adds by region on click', () => {
    expect(addedRegion).toBeFalsy();
    const button = result.find(`#reg_${props.text}`);
    button.simulate('click');
    expect(addedRegion).toBeTruthy();
  });

  it('adds by image on click', () => {
    expect(addedImage).toBeFalsy();
    const btn = result.find(`#image_${props.text}`);
    btn.simulate('click');
    expect(addedImage).toBeTruthy();
  });
});
