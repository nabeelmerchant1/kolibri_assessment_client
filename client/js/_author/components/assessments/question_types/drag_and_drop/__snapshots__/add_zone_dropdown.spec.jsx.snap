// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`drag_and_drop component renders the component as a snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ul
    className="au-c-button-dropdown au-u-ml-sm"
>
    <li>
        <button
            className="au-c-btn au-c-btn--sm au-c-btn--dropdown"
            onClick={[Function]}
        >
            Spec
            <i
                className="material-icons"
            >
                arrow_drop_down
            </i>
        </button>
        <ul
            className="au-c-button-dropdown__dropdown "
        >
            <li>
                <button
                    className="au-c-btn au-c-btn--sm au-c-btn--dropdown-item"
                    id="reg_Spec"
                    onClick={[Function]}
                >
                    <i
                        className="material-icons"
                    >
                        open_in_new
                    </i>
                    by region
                </button>
            </li>
            <li>
                <button
                    className="au-c-btn au-c-btn--sm au-c-btn--dropdown-item"
                    id="image_Spec"
                    onClick={[Function]}
                >
                    <i
                        className="material-icons"
                    >
                        image
                    </i>
                    by image
                </button>
            </li>
        </ul>
    </li>
</ul>,
  "nodes": Array [
    <ul
      className="au-c-button-dropdown au-u-ml-sm"
>
      <li>
            <button
                  className="au-c-btn au-c-btn--sm au-c-btn--dropdown"
                  onClick={[Function]}
            >
                  Spec
                  <i
                        className="material-icons"
                  >
                        arrow_drop_down
                  </i>
            </button>
            <ul
                  className="au-c-button-dropdown__dropdown "
            >
                  <li>
                        <button
                              className="au-c-btn au-c-btn--sm au-c-btn--dropdown-item"
                              id="reg_Spec"
                              onClick={[Function]}
                        >
                              <i
                                    className="material-icons"
                              >
                                    open_in_new
                              </i>
                              by region
                        </button>
                  </li>
                  <li>
                        <button
                              className="au-c-btn au-c-btn--sm au-c-btn--dropdown-item"
                              id="image_Spec"
                              onClick={[Function]}
                        >
                              <i
                                    className="material-icons"
                              >
                                    image
                              </i>
                              by image
                        </button>
                  </li>
            </ul>
      </li>
</ul>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <addZoneDropdown
        active={false}
        addByImage={[Function]}
        addByRegion={[Function]}
        text="Spec"
        toggle={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "active": false,
          "addByImage": [Function],
          "addByRegion": [Function],
          "text": "Spec",
          "toggle": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ul
          className="au-c-button-dropdown au-u-ml-sm"
>
          <li>
                    <button
                              className="au-c-btn au-c-btn--sm au-c-btn--dropdown"
                              onClick={[Function]}
                    >
                              Spec
                              <i
                                        className="material-icons"
                              >
                                        arrow_drop_down
                              </i>
                    </button>
                    <ul
                              className="au-c-button-dropdown__dropdown "
                    >
                              <li>
                                        <button
                                                  className="au-c-btn au-c-btn--sm au-c-btn--dropdown-item"
                                                  id="reg_Spec"
                                                  onClick={[Function]}
                                        >
                                                  <i
                                                            className="material-icons"
                                                  >
                                                            open_in_new
                                                  </i>
                                                  by region
                                        </button>
                              </li>
                              <li>
                                        <button
                                                  className="au-c-btn au-c-btn--sm au-c-btn--dropdown-item"
                                                  id="image_Spec"
                                                  onClick={[Function]}
                                        >
                                                  <i
                                                            className="material-icons"
                                                  >
                                                            image
                                                  </i>
                                                  by image
                                        </button>
                              </li>
                    </ul>
          </li>
</ul>,
        "_debugID": 2,
        "_renderedOutput": <ul
          className="au-c-button-dropdown au-u-ml-sm"
>
          <li>
                    <button
                              className="au-c-btn au-c-btn--sm au-c-btn--dropdown"
                              onClick={[Function]}
                    >
                              Spec
                              <i
                                        className="material-icons"
                              >
                                        arrow_drop_down
                              </i>
                    </button>
                    <ul
                              className="au-c-button-dropdown__dropdown "
                    >
                              <li>
                                        <button
                                                  className="au-c-btn au-c-btn--sm au-c-btn--dropdown-item"
                                                  id="reg_Spec"
                                                  onClick={[Function]}
                                        >
                                                  <i
                                                            className="material-icons"
                                                  >
                                                            open_in_new
                                                  </i>
                                                  by region
                                        </button>
                              </li>
                              <li>
                                        <button
                                                  className="au-c-btn au-c-btn--sm au-c-btn--dropdown-item"
                                                  id="image_Spec"
                                                  onClick={[Function]}
                                        >
                                                  <i
                                                            className="material-icons"
                                                  >
                                                            image
                                                  </i>
                                                  by image
                                        </button>
                              </li>
                    </ul>
          </li>
</ul>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <addZoneDropdown
    active={false}
    addByImage={[Function]}
    addByRegion={[Function]}
    text="Spec"
    toggle={[Function]}
/>,
}
`;
